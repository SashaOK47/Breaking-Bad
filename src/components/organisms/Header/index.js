import React from "react";
import style from "./styles.module.scss";
import { NavLink } from "react-router-dom";
import logo from "../../../assets/images/logo.png";
import burger from "../../../assets/icons/burger.svg";

const Header = () => {
  return (
    <header className={style.header}>
      <div className={style.container}>
        <nav className={style.navigation}>
          <NavLink to="/" className={style.logo}>
            <img src={logo} alt="Logo" />
          </NavLink>
          <NavLink
            end
            to="/characters"
            className={({ isActive }) => `${isActive ? style.active : ""}`}
          >
            <span className={style.catalog}>
              <img src={burger} alt="Burger" />
              <span>Каталог</span>
            </span>
          </NavLink>
        </nav>
      </div>
    </header>
  );
};
export default Header;
