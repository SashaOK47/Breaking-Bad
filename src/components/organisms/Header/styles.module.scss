@import "../../../assets/styles/main.scss";
.header {
  padding: 1rem 0;
  background-color: $white;
  .navigation {
    display: flex;
    justify-content: space-between;
    align-items: center;
    a:last-child {
      padding: 0.5rem 1rem;
      border: 0.1rem solid $colorBgLink;
      border-radius: 0.25rem;
      transition: background-color 0.3s;
      &.active {
        background-color: $colorBgLink;
      }
    }
    .catalog {
      display: flex;
      button {
        background: none;
        margin-left: 0.5rem;
      }
    }
  }
}
