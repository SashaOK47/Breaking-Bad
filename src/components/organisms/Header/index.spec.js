import React from "react";
import Header from "./index";

describe("Header Component", () => {
  it("should render Header component", () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
