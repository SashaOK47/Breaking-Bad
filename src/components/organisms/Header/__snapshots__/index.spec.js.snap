// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render Header component 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <nav
      className="navigation"
    >
      <NavLink
        className="logo"
        to="/"
      >
        <img
          alt="Logo"
          src="test-file-stub"
        />
      </NavLink>
      <NavLink
        className={[Function]}
        end={true}
        to="/characters"
      >
        <span
          className="catalog"
        >
          <img
            alt="Burger"
            src="test-file-stub"
          />
          <span>
            Каталог
          </span>
        </span>
      </NavLink>
    </nav>
  </div>
</header>
`;
